models/product-data/src/transformers.ts

```diff
@@ -0,0 +1,103 @@
+import { CategoryReference } from '@commercetools/platform-sdk';
+import { LocalizedString } from '@commercetools-test-data/commons';
+import {
+  buildField,
+  buildFields,
+  Transformer,
+} from '@commercetools-test-data/core';
+import type {
+  TCategoryOrderHintGraphql,
+  TProductData,
+  TProductDataGraphql,
+  TProductDataRest,
+} from './types';
+
+const transformers = {
+  default: Transformer<TProductData, TProductData>('default', {
+    buildFields: [
+      'name',
+      'categories',
+      'description',
+      'slug',
+      'metaTitle',
+      'metaDescription',
+      'metaKeywords',
+      'masterVariant',
+      'variant',
+      'variants',
+      'allVariants',
+      'searchKeywords',
+      'searchKeyword',
+      'categoriesRef',
+    ],
+  }),
+  rest: Transformer<TProductData, TProductDataRest>('rest', {
+    replaceFields: ({ fields }) => {
+      const { categories } = fields;
+
+      const categoryReferences: Array<CategoryReference> = buildFields(
+        categories
+      ).map((category) => ({
+        id: category.id,
+        typeId: 'category',
+      }));
+
+      /**
+       * We cannot use `replaceFields` in conjunction with `buildFields`,
+       * so we need to explicity build fields where necessary.
+       */
+      return {
+        // Un-built fields with no model dependencies
+        ...fields,
+        // These have model dependencies and must be built
+        name: buildField(fields.name),
+        description: buildField(fields.description),
+        slug: buildField(fields.slug),
+        metaTitle: buildField(fields.metaTitle),
+        metaDescription: buildField(fields.metaDescription),
+        metaKeywords: buildField(fields.metaKeywords),
```

Comment:
Do we need the `rest` parameter here?

Reply:
Works for me, it's more explicit. See 4cc8c8796537c1b78b0b145f80b28b61b3ac83c3